code/modules/admin/admin.dm

```diff
@@ -2734,6 +2734,28 @@ var/global/noir = 0
 						logTheThing("admin", usr, null, "used Farty Party secret")
 						logTheThing("diary", usr, null, "used Farty Party secret", "admin")
 
+					if("give-legs")
+						if(src.level >= LEVEL_ADMIN)
+							var/list/atoms_to_leg = list()
+							var/pathname = input("Path of the things you want to leg like (/obj/item/paper_bin)","path:", null) as null|text
+							if(!pathname)
+								return
+							if (alert("Do you really wanna give things legs?", "LEG LEG", "Sure thing!", "Not really.") == "Sure thing!")
+								for(var/atom/movable/A in world) //Build the god forsaken list
+									if(istype(A, text2path(pathname)))
+										atoms_to_leg += A
```

Comment:
What is the purpose of building this list? It doesn't look like you're doing anything with it and it just needlessly bumps up your time complexity from O(n) to O(n + m) where n is the number of atoms in world and m is the number of atoms in your list.

Also it could be a good idea to add functionality for easily undoing this.